BlackBird not support fluent setter

# Environment
* jdk 11
* jackson 2.12.0 with BlackBird
# Reproduce case
 define a model with fluent setter  
  ```java
public class BlackBirdBug {
    public static class Model {
        private int value = 10;

        public int getValue() {
            return value;
        }

        public Model setValue(int value) {
            this.value = value;

            return this;
        }
    }

    public static void main(String[] args) throws JsonProcessingException {
        ObjectMapper mapper = new ObjectMapper()
            .registerModule(new BlackbirdModule());
        String json = mapper.writeValueAsString(new Model());
        System.out.println(json);

        Model model = mapper.readValue(json, Model.class);
        System.out.println(model.getValue());
    }
}
  ```  

will throw exception:
```
Caused by: java.lang.invoke.LambdaConversionException: Type mismatch for instantiated parameter 1: int is not a subtype of class java.lang.Object
	at java.base@11.0.6/java.lang.invoke.AbstractValidatingLambdaMetafactory.checkDescriptor(AbstractValidatingLambdaMetafactory.java:308)
	at java.base@11.0.6/java.lang.invoke.AbstractValidatingLambdaMetafactory.validateMetafactoryArgs(AbstractValidatingLambdaMetafactory.java:294)
	at java.base@11.0.6/java.lang.invoke.LambdaMetafactory.metafactory(LambdaMetafactory.java:328)
	at com.fasterxml.jackson.module.blackbird.deser.BBDeserializerModifier.createSetter(BBDeserializerModifier.java:221)
	at com.fasterxml.jackson.module.blackbird.deser.BBDeserializerModifier.nextProperty(BBDeserializerModifier.java:179)
	at com.fasterxml.jackson.module.blackbird.deser.BBDeserializerModifier.findOptimizableProperties(BBDeserializerModifier.java:126)
	... 14 more
```  

we are trying switch from AfterBurner to BlackBird, because AfterBurner not support `ToStringSerializer` via `@JsonSerialize` in `boolean` oroperty

--- a/blackbird/src/main/java/com/fasterxml/jackson/module/blackbird/deser/BBDeserializerModifier.java
+++ b/blackbird/src/main/java/com/fasterxml/jackson/module/blackbird/deser/BBDeserializerModifier.java
@@ -222,7 +222,7 @@ public class BBDeserializerModifier extends BeanDeserializerModifier
                 MethodHandles.lookup(),
                 "accept",
                 MethodType.methodType(iface, thunkType),
-                MethodType.methodType(void.class, Object.class, Object.class),
+                MethodType.methodType(void.class, Object.class, valueType),
                 trampoline,
                 MethodType.methodType(void.class, Object.class, valueType))
             .getTarget().invoke(builtThunk));

--- a/blackbird/src/test/java/com/fasterxml/jackson/module/blackbird/deser/BasicDeserialize123Test.java
+++ b/blackbird/src/test/java/com/fasterxml/jackson/module/blackbird/deser/BasicDeserialize123Test.java
@@ -1,4 +1,4 @@
-package com.fasterxml.jackson.module.blackbird.failing;
+package com.fasterxml.jackson.module.blackbird.deser;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
 

